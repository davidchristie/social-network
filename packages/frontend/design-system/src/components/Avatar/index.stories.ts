import Avatar from ".";
import addComponentStory from "../../utilities/storybook/addComponentStory";

addComponentStory(Avatar, {
  children: undefined,
});
