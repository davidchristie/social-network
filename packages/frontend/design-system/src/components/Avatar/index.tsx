import MaterialAvatar from "@material-ui/core/Avatar";
import React from "react";

import Icon from "../Icon";

interface Props {
  children?: React.ReactNode;
  image?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
}

const Avatar: React.SFC<Props> = ({
  children,
  image,
  ...attributes
}) => {
  return (
    <MaterialAvatar src={image} {...attributes}>
      {children}
      {!children && !image && (
        <Icon>account_circle</Icon>
      )}
    </MaterialAvatar>
  );
};

export default Avatar;
