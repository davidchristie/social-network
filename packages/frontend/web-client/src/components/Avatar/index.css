.Avatar {
  background: white;
  border: 1px solid black;
  border-radius: 50%;
}

.Avatar.small {
  height: 35px;
  width: 35px;
}

.Avatar.large {
  height: 70px;
  width: 70px;
}
