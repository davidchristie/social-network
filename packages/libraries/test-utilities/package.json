{
  "name": "test-utilities",
  "private": true,
  "scripts": {
    "build": "rimraf dist && tsc"
  },
  "version": "1.0.0"
}
