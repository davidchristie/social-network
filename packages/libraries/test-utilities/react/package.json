{
  "main": "../dist/react/index.js",
  "name": "test-utilities/react",
  "private": true,
  "version": "1.0.0"
}
