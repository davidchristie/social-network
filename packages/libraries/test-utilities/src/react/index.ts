export { default as describeMountedElement } from "./describeMountedElement";
export { default as describeWithCustomClass } from "./describeWithCustomClass";
export { default as describeWithNoProps } from "./describeWithNoProps";
export { default as describeWithProps } from "./describeWithProps";
export { default as itMatchesSnapshot } from "./itMatchesSnapshot";
