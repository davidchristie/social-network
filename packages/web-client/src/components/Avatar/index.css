.Avatar {
  border-radius: 50%;
}

.Avatar.small {
  height: 35px;
  width: 35px;
}

.Avatar.large {
  height: 70px;
  width: 70px;
}
