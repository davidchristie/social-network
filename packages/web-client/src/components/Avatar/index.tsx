import React from "react";

import "./index.css";

interface Props {
  image: string;
  onClick?: (event: React.MouseEvent<HTMLImageElement>) => void;
  size: "small" | "large";
}

export default class Avatar extends React.Component<Props> {
  public render () {
    return (
      <img
        className={`Avatar ${this.props.size}`}
        onClick={this.props.onClick}
        src={this.props.image}
      />
    );
  }
}
