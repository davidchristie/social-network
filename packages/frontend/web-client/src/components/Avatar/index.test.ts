import { describeWithProps } from "test-utilities/react";

import Avatar from ".";

function describeSize (size: string) {
  describeWithProps(size, Avatar, { size });
}

describe("Avatar component", () => {
  describeSize("small");

  describeSize("large");
});
