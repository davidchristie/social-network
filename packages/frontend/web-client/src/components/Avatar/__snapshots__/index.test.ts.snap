// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component large matches snapshot 1`] = `
<img
  className="Avatar large"
/>
`;

exports[`Avatar component small matches snapshot 1`] = `
<img
  className="Avatar small"
/>
`;
